spec/fixtures/mineunit_extensions.lua

```diff
@@ -0,0 +1,111 @@
+
```

Comment:
What does this file do? Why is it needed?

Reply:
> These unittest files would fit better into a separate dependency.

From standpoint of runtime, yeah, could save some bandwidth as these files are only needed for testing and developing.

@S-S-X do you have a suggestion for clean splitting that still keeps devs aware to add and test. Also that would work well with GH-Workflows?
Would we have a collection of `mineunit_modname` repos under one organization or smth similar?

Reply:
It is very much possible and there's clear benefits for having tests in separate repository... but only when project is large enough to require its own dedicated project space (dedicated GitHub organization for spectator_mode).

For small and clean projects like Minetest engine or Minetest mods separate repository for tests will just make it confusing, harder to maintain full picture and way harder to maintain tests. I guess you can already imagine a lot of reasons for this?

_Going for single run tests that will not be maintained?_ Then sure separate repo is definitely better, easier to throw away.
_Going to actually use and maintain regression tests?_ Keep test specifications in same repo, subdirectory approach is pretty much standard and there's good reasons why small projects utilize this.

Then of course I'll add one opinion I do have about tests and contributions.
Pull requests that add or modify both tests and main code should be cleaned up before mering in a way that:
* No Squash.
* Separate commits for mod code and tests.
* Clean enough to be acceptable in main development branch history.
* Just repeating for clarification: affects only situation where *both* mod code *and* tests are modified within same source branch.

Reply:
> These unittest files would fit better into a separate dependency.

Very much agreed with these exact "workaround" libraries.
These, being temporary, could go into separate repository and pulled from there with checkout action.

> The methods in this file are needed by the unit tests bc mineunit did/does not have them implemented yet.

Flags `--fetch-core 5.5.0 --engine-version 5.5.0` works already in github actions but I should add easy to use parameter `engine-version: 5.5.0` for github actions yaml. Other stuff not yet merged in.

Reply:
> > These unittest files would fit better into a separate dependency.
> 
> Very much agreed with these exact "workaround" libraries. These, being temporary, could go into separate repository and pulled from there with checkout action.

Having exact state frozen to the repo it is in, especially at this size, seems reasonable to me. Both this whole spec directory and this specific file.

Reply:
> I like that, may I also recommend to prefix commit names with 'mu ' or 'mineunit ' for easier filtering

Or just generic "fix bug xyz" (mod code commit) + "tests for xyz" (whatever test related stuff). It could mention exact test framework but should be enough to just separate tests from code, no matter how.

For me it is mostly for easy to navigate history with important related stuff on clearly connected and clean timeline (I'm not even really against having code and tests in single commit but separate commits feels cleaner, easier to revert if needed, easier to merge elsewhere and simpler to review).